### PR TITLE
[MM-22368] Log Channel Moderation events

### PR DESCRIPTION
**Pull request by fm2munsh**
#### Summary
- Adds logging for when a channel moderation setting is enabled / disabled or a channel permission scheme is disabled or enabled
- Sample Logging: ```{"level":"info","ts":1583945483.2817202,"caller":"app/channel.go:858","msg":"manage_members: enabled for Members in channel id: 6fo1o1335bdkbxi61kgam4fkgh"}
{"level":"info","ts":1583945486.10149,"caller":"app/channel.go:844","msg":"Permission scheme deleted for channel id: 6fo1o1335bdkbxi61kgam4fkgh"}
{"level":"info","ts":1583945495.340341,"caller":"app/channel.go:817","msg":"Permission scheme created for channel id: 6fo1o1335bdkbxi61kgam4fkgh"}
{"level":"info","ts":1583945495.350199,"caller":"app/channel.go:860","msg":"create_reactions: disabled for Members in channel id: 6fo1o1335bdkbxi61kgam4fkgh"}
{"level":"info","ts":1583945495.350254,"caller":"app/channel.go:860","msg":"manage_members: disabled for Members in channel id: 6fo1o1335bdkbxi61kgam4fkgh"}
{"level":"info","ts":1583945499.160875,"caller":"app/channel.go:858","msg":"create_reactions: enabled for Members in channel id: 6fo1o1335bdkbxi61kgam4fkgh"}
{"level":"info","ts":1583945501.993239,"caller":"app/channel.go:844","msg":"Permission scheme deleted for channel id: 6fo1o1335bdkbxi61kgam4fkgh"}``` 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22368

### CODE REVIEW

**Comment by hahmadia** [comment]
can you also include the channel name? thanks

File: app/channel.go [hunk 1]
```diff
@@ -753,6 +753,7 @@ func (a *App) PatchChannelModerationsForChannel(channel *model.Channel, channelM
 		if _, err = a.CreateChannelScheme(channel); err != nil {
 			return nil, err
 		}
+		mlog.Info("Permission scheme created for channel id: " + channel.Id + " with name: " + channel.Name)
```
Comment (mkraft) [inline app/channel.go]:
```suggestion
		mlog.Info("Permission scheme created.", mlog.String("channel_id", channel.Id), mlog.String("channel_name", channel.Name)
```

**Comment by fm2munsh** [comment]
@srkgupta this is good for QA now

File: app/channel.go [hunk 2]
```diff
@@ -769,6 +770,25 @@ func (a *App) PatchChannelModerationsForChannel(channel *model.Channel, channelM
...
+			if *channelModerationPatch.Roles.Guests {
+				mlog.Info("Permission enabled for guests.", mlog.String("permission", permissionModified), mlog.String("channel_id", channel.Id), mlog.String("channel_name", channel.Name))
+			} else {
+				mlog.Info("Permission disabled for guests.", mlog.String("permission", permissionModified), mlog.String("channel_id", channel.Id), mlog.String("channel_name", channel.Name))
+			}
+		}
+
+		if channelModerationPatch.Roles.Members != nil && utils.StringInSlice(permissionModified, model.ChannelModeratedPermissionsChangedByPatch(memberRole, memberRolePatch)) {
+			if *channelModerationPatch.Roles.Members {
+				mlog.Info("Permission disabled for members.", mlog.String("permission", permissionModified), mlog.String("channel_id", channel.Id), mlog.String("channel_name", channel.Name))
```
Comment (fm2munsh) [inline app/channel.go]:
```suggestion
				mlog.Info("Permission enabled for members.", mlog.String("permission", permissionModified), mlog.String("channel_id", channel.Id), mlog.String("channel_name", channel.Name))
```

**Comment by fm2munsh** [comment]
@srkgupta I just had a typo in the logging - it should be fixed now

**Pull request merged**